core/common/src/main/java/alluxio/security/authentication/AbstractSaslClientHandler.java

```diff
@@ -85,10 +88,11 @@ public SaslMessage handleMessage(SaslMessage message) throws SaslException {
   }
 
   @Override
-  public void close() {
+  public synchronized void close() {
     if (mSaslClient != null) {
       try {
         mSaslClient.dispose();
+        mSaslClient = null;
```

Comment:
do we want to keep this non-null if there is an issue with `dispose`? Basically, should we set it to null in finally or here?

Reply:
We don't want to keep it set if dispose fails. So in finally is better.